Tools/WinMLRunner/OutputHelper.h

```diff
@@ -9,65 +8,162 @@
+            {
+                PrintFeatureDescriptorInfo(outputFeature);
+            }
+            std::cout << "=================================================================" << std::endl;
+            std::cout << std::endl;
+        }
+    }
+
+    void PrintFeatureDescriptorInfo(const ILearningModelFeatureDescriptor &descriptor) const
+    {
+        if (!m_silent)
+        {
+            //IMPORTANT: This learningModelFeatureKind array needs to match the "enum class 
+            //LearningModelFeatureKind" idl in Windows.AI.MachineLearning.0.h
+            const std::string learningModelFeatureKind[] =
+            {
+                "Tensor",
+                "Sequence",
+                "Map",
+                "Image",
+            };
+            std::wstring name(descriptor.Name());
+            std::wcout << "Name: " << name << std::endl;
+            std::wcout << "Feature Kind: " << FeatureDescriptorToString(descriptor) << std::endl;
+            std::cout << std::endl;
+        }
     }
 
-    void PrintCPUTimes(Profiler<WINML_MODEL_TEST_PERF> &profiler, UINT iterations)
+    void PrintHardwareInfo() const
     {
-         m_CPULoadTime = profiler[LOAD_MODEL].GetAverage(CounterType::TIMER);
-         m_CPUBindTime = profiler[BIND_VALUE].GetAverage(CounterType::TIMER);
-         m_CPUEvalTime = profiler[EVAL_MODEL].GetAverage(CounterType::TIMER);
-         m_CPUEvalMemoryUsage = profiler[EVAL_MODEL].GetAverage(CounterType::WORKING_SET_USAGE);
-
-        std::cout << std::endl;
-        std::cout << "CPU Time Averages (iterations = " << iterations << "):" << std::endl;
-        std::cout << "  Load: " << m_CPULoadTime << " ms" << std::endl;
-        std::cout << "  Bind: " << m_CPUBindTime << " ms" << std::endl;
-        std::cout << "  Evaluate: " << m_CPUEvalTime << " ms" << std::endl;
-        std::cout << "  Total time: " << m_CPULoadTime + m_CPUBindTime + m_CPUEvalTime << " ms" << std::endl;
-        std::cout << "  Working Set Memory usage (evaluate): " << m_CPUEvalMemoryUsage << " MB" << std::endl;
-        std::cout << std::endl;
+        if (!m_silent)
+        {
+            std::cout << "WinML Runner" << std::endl;
+
+            com_ptr<IDXGIFactory6> factory;
+            CreateDXGIFactory1(__uuidof(IDXGIFactory6), factory.put_void());
+            com_ptr<IDXGIAdapter> adapter;
+            factory->EnumAdapters(0, adapter.put());
+            DXGI_ADAPTER_DESC description;
+            if (SUCCEEDED(adapter->GetDesc(&description)))
+            {
+                std::wcout << L"GPU: " << description.Description << std::endl;
+                std::cout << std::endl;
+            }
+        }
     }
 
-    void PrintGPUTimes(Profiler<WINML_MODEL_TEST_PERF> &profiler, UINT iterations)
+    void PrintResults(const Profiler<WINML_MODEL_TEST_PERF> &profiler, uint32_t numIterations, DeviceType deviceType, InputBindingType inputBindingType, InputDataType inputDataType) const
```

Comment:
We now have a single PrintResults() method, no matter which configuration is used.